clustered/common/src/main/java/org/ehcache/clustered/common/internal/exceptions/ClusteredEhcacheException.java

```diff
@@ -29,4 +30,14 @@ public ClusteredEhcacheException(String message, Throwable cause) {
   public ClusteredEhcacheException(Throwable cause) {
     super(cause);
   }
+
+  /**
+   * Makes a copy of this {@code ClusteredEhcacheException}, retaining the type,
+   * in the current context.  This method is used to prepare exceptions originating in
+   * tbe server to be re-thrown in the client.
+   *
+   * @return a new {@code ClusteredEhcacheException} of {@code this} subtype with {@code this}
+   *      instance as the cause
+   */
+  public abstract ClusteredEhcacheException copyInContext();
```

Comment:
I'm not enamoured with this method name, it doesn't seem to have the right connotations to me... but damned if I can think of a better one.